src/Microsoft.Data.Sqlite.Core/SqliteParameter.cs

```diff
@@ -95,7 +95,7 @@ public SqliteParameter(string name, SqliteType type, int size, string sourceColu
         /// <value>The SQLite type of the parameter.</value>
         /// <remarks>Due to SQLite's dynamic type system, parameter values are not converted.</remarks>
         /// <seealso href="http://sqlite.org/datatype3.html">Datatypes In SQLite Version 3</seealso>
-        public virtual SqliteType SqliteType { get; set; } = SqliteType.Text;
+        public virtual SqliteType SqliteType { get; set; } = SqliteType.Undefined;
```

Comment:
Use a `SqliteType?` field instead. If they read the value without setting it, we should infer the type from `_value`.